chat-android/src/main/java/com/ably/chat/Typing.kt

```diff
@@ -97,7 +81,29 @@ public fun Typing.asFlow(): Flow<TypingEvent> = transformCallbackAsFlow {
 /**
  * Represents a typing event.
  */
-public data class TypingEvent(val currentlyTyping: Set<String>)
+public data class TypingEvent(
+    /**
+     * The set of user clientIds that are currently typing.
+     */
+    val currentlyTyping: Set<String>,
+
+    /**
+     * The change that caused this event.
+     */
+    val change: TypingChange,
+) {
+    public data class TypingChange(
```

Comment:
I don't really like nested data classes inside data class, it doesn't feel very Kotlin-idiomatic. But we have to replace it with an interface anyway. I would prefer:

```kotlin
interface TypingEvent {
  public val currentlyTyping: Set<String>
  public val change: TypingEventChange
}

interface TypingEventChange {
  public val clientId: String
  public val type: TypingEventType
}
```

But it this will also work, but I think "Typing" prefix is  unnecessary in this case

```kotlin
interface TypingEvent {
  public val currentlyTyping: Set<String>
  public val change: Change
 
  interface Change {
    public val clientId: String
    public val type: TypingEventType
  }
}

```

Reply:
Fixed https://github.com/ably/ably-chat-kotlin/pull/117/commits/00a0e7d3b66ac64a87b69cdfc872cd601f0efe04